src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940ValidationUtils.java

```diff
@@ -0,0 +1,404 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.module.swiftiso20022.utils;
+
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+import org.wso2.carbon.module.swiftiso20022.model.ErrorModel;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Validation utils for MT940.
+ */
+public class MT940ValidationUtils {
+
+    /**
+     * Method to validate the C1 rule MT940 format.
+     * C1 rule - :86: should be preceded by :61:
+     *
+     * @param lines  Lines in MT940 format
+     * @return    Result of C1 rule validation
+     */
+    public static boolean validateC1Rule(String[] lines) {
+
+        for (int itr = 0; itr < lines.length; itr++) {
+            if (lines[itr].startsWith(ConnectorConstants.MT940_INFORMATION)) {
+                return lines[itr - 1].startsWith(ConnectorConstants.MT940_STATEMENT_LINE);
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Method to validate the C2 rule MT940 format.
+     * C2 rule - :60:, :62:, :64: and :65: should have same currency code
+     *
+     * @param lines  Lines in MT940 format
+     * @return    Result of C2 rule validation
+     */
+    public static boolean validateC2Rule(String[] lines) {
+        String openBalance = null;
+        String closeBalance = null;
+        String closeAvailBalance = null;
+        String forwardAvailBalance = null;
+
+        for (String line : lines) {
+            if (line.startsWith(ConnectorConstants.MT940_OPENING_BAL)) {
+                openBalance = line;
+            } else if (line.startsWith(ConnectorConstants.MT940_CLOSING_BAL)) {
+                closeBalance = line;
+            } else if (line.startsWith(ConnectorConstants.MT940_CLOSING_AVAIL_BAL)) {
+                closeAvailBalance = line;
+            } else if (line.startsWith(ConnectorConstants.MT940_FORWARD_AVAIL_BAL)) {
+                forwardAvailBalance = line;
+            }
+        }
+
+        if (!openBalance.substring(13, 15).equals(closeBalance.substring(13, 15))) {
```

Comment:
Won't this throw NPE?

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/8d6b87e8988955fbe9679e7afa9d4285b0ff0ac8